src/main/java/com/kodilla/ecommercee/controller/CartController.java

```diff
@@ -1,4 +1,41 @@
 package com.kodilla.ecommercee.controller;
 
+import com.kodilla.ecommercee.dto.CartDto;
+import com.kodilla.ecommercee.dto.OrderDto;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+
+@CrossOrigin(origins = "*")
+@RestController
+@RequestMapping("v1/ecommercee/")
 public class CartController {
-}
+
```

Comment:
Hmm mam kilka uwag:
1. Nazwa kontrolera nie wydaje mi sie zbyt intuicyjna. Ponieważ nie określa ona specyfikacji controlera . "v1/ecommercee" mówi tylko że jest to controler wersji pierwszej projektu ecommerce. Gdy bys dał nazwę "v1/ecomerce/cart" było by odrazu wiadomo że controler dotyczy koszyka.

2. Widzę ,że  używasz adnotacji @PathVariable której argumenty podajemy w środkowej części url jako część value endpointa, nie wiem czy łatwiejszym rozwiązaniem nie będzie używanie wszędzie @RequestParam ponieważ wtedy argumenty zawsze "doklejamy" na końcu url po znaku "?", dzięki temu twoja wartosć value będzie duza prostsza w zapisie. Mam na myśli coś takiego:
TWOJA WERSJA:
@PostMapping(value = "{userId}/cart/{itemId}")
    public  void addItemToCard(@PathVariable Long userId, @PathVariable Long itemId,  @RequestParam Long itemQuantity) {
    }
MOJA WERSJA:
@PostMapping(value = "addItemToCard")
    public  void addItemToCard(@RequestParam Long userId, @RequestParam Long itemId,  @RequestParam Long itemQuantity) {
    }
Wydaje mi się ,że zmienienie tych wartosci endpointów  będzie skutkować lepszą czytelnością kodu.

Reply:
<a href="https://ibb.co/86b81BQ"><img src="https://i.ibb.co/SRcvFxZ/Screen-Shot-07-25-19-at-03-30-PM.png" alt="Screen-Shot-07-25-19-at-03-30-PM" border="0"></a>

Jeżeli dołączymy dokumentacje wszystko staje się czytelne.

1. Zwróć uwagę żę "v1/ecommercee" nie jest pełną nazwą (patrz dokumentacja).
2. Moim zamysłem było zwykłe drzewo (struktura katalogów). 
    O ile produkt np może mieć bezpośredni adres
     v1/eccommerce/product

to wydawało mi się że naturalne będzie że koszyk jest  "w katalogu" usera. Przecież każdy user ma swój koszyk. Jest on zawsze, i tak naprawdę ma stałe ID? bo przecież koszyk nie znika. Możemy go wyczyścić ale on dalej jest tym samym koszykiem!

Wzorowałem się na trello  gdzie: 
 usuwanie, dodawanie, uaktualnienie Card ma ten sam adres różni się tylko metodami post, get itp...    
/card/{id}


https://blog.mwaysolutions.com/2014/06/05/10-best-practices-for-better-restful-api/

Już w pierwszym punkcie jest pokazane jak nie powinno się robić.

Reply:
Jako że dwa  controllery już poszły i zostały zatwierdzone.  Poszedłem tym samym tropem aby iść dalej a nie stać w miejscu! :)